dataflux_pytorch/benchmark/checkpointing/simulated/multi_node_benchmark.py

```diff
@@ -49,56 +44,64 @@ def init_processes() -> int:
-    model = SimpleModel(model_size, padding_size)
-
-    dummy_input = torch.randn(100, model_size)
-    _ = model(dummy_input)
-
-    full_state_dict = model.state_dict()
-    for i, tensor in enumerate(model.dummy_tensors):
-        full_state_dict[f'dummy_tensor_{i}'] = tensor
 
     benchmark_strategy = BenchmarkStrategy(
-        project=project, path=filepath, model=model)
-
-    distributed_state_dict = {f"{key}_shard_{rank}": split_tensor(
-        tensor, world_size, rank) for key, tensor in full_state_dict.items()}
-
-    save_checkpoint_times = time_checkpoint_operation(
-        benchmark_strategy, distributed_state_dict, filepath, sample_size, 'save')
-    load_checkpoint_times = time_checkpoint_operation(
-        benchmark_strategy, distributed_state_dict, filepath, sample_size, 'load')
+        project=project,
+        path=filepath,
+    )
+
+    state_dict = dict()
+    for i in range(padding_size):
+        if i % world_size == rank:
+            state_dict[f'dummy_tensor_{i}'] = torch.randn(layer_size, 1000)
+    dist.barrier()
+
+    save_checkpoint_times = time_checkpoint_operation(benchmark_strategy,
+                                                      state_dict, filepath,
+                                                      sample_count, 'save',
+                                                      rank, world_size,
+                                                      padding_size, layer_size)
+
+    load_checkpoint_times = time_checkpoint_operation(benchmark_strategy,
+                                                      state_dict, filepath,
+                                                      sample_count, 'load',
+                                                      rank, world_size,
+                                                      padding_size, layer_size)
 
     if rank == 0:
-        print("######################")
-        print(
-            f"Time taken to save checkpoint: {statistics.mean(save_checkpoint_times):.4f} seconds")
-        print(
-            f"Time taken to load checkpoint: {statistics.mean(load_checkpoint_times):.4f} seconds")
-        total_distributed_size_bytes = sum(get_tensor_size_bytes(
-            tensor) for tensor in distributed_state_dict.values())
-        print(
-            f"Size of distributed tensors (rank {rank}): {format_size(total_distributed_size_bytes)}")
-        print(
-            f"Total size of all tensors (rank {rank}): {format_size(total_distributed_size_bytes * world_size)}")
+        print(f"Time taken to save checkpoint:\
+                {statistics.mean(save_checkpoint_times):.4f} seconds")
+        print(f"Time taken to load checkpoint:\
+                 {statistics.mean(load_checkpoint_times):.4f} seconds")
+
+        tensor_size_per_instance = 1000 * layer_size * state_dict[
+            f'dummy_tensor_0'].element_size()
+        total_tensor_count = padding_size // world_size  # Total tensors per rank
```

Comment:
probably should name this `tensors_per_rank` since it's not the total.

Reply:
Done